doc/examples/test_serialization.cpp

```diff
@@ -80,11 +85,30 @@ auto main() -> int {
     return 1;
   }
 
-  Memory::buffer = (uint8_t*)mmap(
-    (void*)0x10'0000'0000, 2 * 65536,
-    PROT_READ | PROT_WRITE | PROT_EXEC,
-    MAP_FIXED | MAP_PRIVATE | MAP_ANONYMOUS, -1, 0
-  );
+  #ifndef _WIN32
+  
+    Memory::buffer = (uint8_t*)mmap(
+      (void*)0x10'0000'0000, 2 * 65536,
+      PROT_READ | PROT_WRITE | PROT_EXEC,
+      MAP_FIXED | MAP_PRIVATE | MAP_ANONYMOUS, -1, 0
+    );
+    
+  #else
+  
+    HANDLE handle = CreateFileMappingW(INVALID_HANDLE_VALUE, nullptr, PAGE_READWRITE, 0, 2 * 65536, nullptr);
```

Comment:
What are the pros and cons of using `CreateFileMappingW` versus `CreateFileMapping` here? I'm not sure what best practices are on Windows these days.

Reply:
We're not naming this memory area (the last argument is null), so W vs A vs neither makes no difference whatsoever.

Reply:
My hazy recollection is that there's different ways to compile code with MSVC and `CreateFileMapping` will automatically use whichever of `CreateFileMappingA` or `CreateFileMappingW` is appropriate for the current mode. If that's not a concern here, that's fine; if that's only an issue when building Win95-era code, that's fine too. I just thought I should ask.

Reply:
In `CreateFileMappingW`, the last argument will be a string of wide-characters; in `CreateFileMappingA`, it will be a string of 8-bit characters. However, since the argument in `nullptr`, it doesn't matter.

However, this should be `NULL` instead of `nullptr`, right? `libco` is also a valid C library, but I'm not sure how C++ deals with raw `NULL`.

Reply:
Yes, CreateFileMapping will indeed map to A or W depending on some preprocessor defines.

The A versions internally convert their string arguments to utf16 and call W. You can mix them just fine. And in this case, the string argument is null, so said conversion is a noop.


That file is example code written in C++, not part of libco itself, so nullptr is correct. There's plenty of other C++-only constructions, like functions in structs and trailing return types, across that file.

Reply:
Thanks for explaining, Alcaro.